packages/api/src/utils/stripe.js

```diff
@@ -256,7 +258,26 @@ export class StripeCustomersService {
       name: options?.name,
       email: options?.email
     })
-    await this.userCustomerService.upsertUserCustomer(user.id.toString(), createdCustomer.id)
+
+    try {
+      await this.userCustomerService.insertUserCustomer(user.id.toString(), createdCustomer.id)
+    } catch (error) {
+      // due to potential race conditions, we want to ensure we only return an actual associated stripe customer
+      // and remove stripe customers that we fail to associate
+      if (hasOwnProperty(error, 'code') && error.code === 'CONSTRAINT_ERROR_DB') {
+        // no need to block during deletion
+        this.stripe.customers.del(createdCustomer.id)
```

Comment:
I don't think it's important to have this clean it up in stripe. We can detect them later on a cron job like you had been implementing.

If we want to take some action here to make that easier, we could write into a table on our side of 'known orphaned customer ids' so we can use that db later to drive a delete job.

but tl;dr I think it would be fine to not put cleanup logic here, and instead just document that it's a known possible side effect of a known race condition, and that it shouldn't affect the running behavior of the app (e.g. no tests fail because of it)